src/sidebar/polling/client.js.html

```diff
@@ -0,0 +1,118 @@
+<script>
+  
+  /*<editor-fold desc="Polling helper">*/
+  
+  /**
+   * Create polling instance
+   *
+   * @param {function(result)} callback
+   * @param {function(error)} errorHandler
+   *
+   * @constructor
+   */
+  var Polling = function (callback, errorHandler) {
+    this._callback = callback;
+    this._callbackError = errorHandler;
+    
+    this._running = false;
+    this._runTimeout = null;
+    this._runningPromise = null;
+  };
+  
+  
+  /**
+   * Start polling
+   */
+  Polling.prototype.start = function() {
+    // Run only once
+    if (this._running) return;
+    this._running = true;
+    
+    clearTimeout(this._runTimeout);
+    !this._runningPromise && this._run();
+  };
+  
+  /**
+   * Stop polling
+   *
+   * This will immediately block all pending polling:
+   * Neither onSuccess not onFailure will be called again
+   */
+  Polling.prototype.stop = function() {
+    if (!this._running) return;
+    
+    this._running = false;
+    clearTimeout(this._runTimeout);
+  };
+  
+  
+  /*<editor-fold desc="Private functions">*/
+  
+  /**
+   * run one poll,
+   * prepare the next one
+   */
+  Polling.prototype._run = function () {
+    if (!this._running) return;
+    
+    this._runningPromise = this._runPollingWithTimeOut()
+      .then(function(res) {
+        if (!this._running) return;
+        
+        return this._callback(res);
+      }.bind(this))
+      .catch(function(err) {
+        if (!this._running) return;
+        
+        return this._callbackError(err);
+      }.bind(this))
+      
+      // Prepare next call
+      .then(function() {
+        this._runningPromise = null;
+        if (!this._running) return;
+        
+        this._runTimeout = setTimeout(this._run.bind(this), Polling.POLLING_DELAY);
```

Comment:
It would make sense to add a visibility.js library, to activate polling only when the UI is visible.
https://github.com/ai/visibilityjs

Reply:
See this post on the G+ community:
https://plus.google.com/u/0/+EricKoleda/posts/i8UyaxYmi7p